src/DynamoPackages/PackageLoader.cs

```diff
@@ -397,29 +374,61 @@ public void LoadPackages(IEnumerable<Package> packages)
                 }
             }
 
-            var assemblies = packageList
+            var assemblies = packages
                 .SelectMany(p => p.LoadedAssemblies.Where(y => y.IsNodeLibrary))
                 .Select(a => a.Assembly)
                 .ToList();
+
             OnPackagesLoaded(assemblies);
         }
 
-        public void LoadCustomNodesAndPackages(LoadPackageParams loadPackageParams, CustomNodeManager customNodeManager)
+        private void LoadNewPackages(LoadPackageParams loadPackageParams)
         {
-            foreach (var path in loadPackageParams.Preferences.CustomPackageFolders)
+            foreach (var path in loadPackageParams.NewPaths)
             {
-                customNodeManager.AddUninitializedCustomNodesInPath(path, false, false);
-                if (!packagesDirectories.Contains(path))
+                var packagesDirectory = pathManager.PackagesDirectories.Where(x => x.StartsWith(path)).FirstOrDefault();
```

Comment:
can you explain the why "startsWith" is used here? What case are you trying to cover that is not covered by contains?

Reply:
The new path added by the user can be a subdirectory of an already existing package directory or a path to a DLL that is a subpath of an existing package directory.

Reply:
okay, and in those cases we don't want to readd that package - correct?

Reply:
Correct, I don't want to process the child directory again, just the parent directory, which should take care of all child paths.